pkg/iter/sample_iterator.go

```diff
@@ -3,6 +3,7 @@ package iter
 import (
 	"container/heap"
 	"context"
+	"go.uber.org/atomic"
```

Comment:
Why use this instead of https://pkg.go.dev/sync/atomic?

Reply:
we have them both in the codebase, I went by convention as  the uber one is more commonly used, whereas "sync/atomic" is generally is in vendored code. 
Do you have a preference for one over the other?

Reply:
No, just wondering, and there are some benefits to uber's library: https://pkg.go.dev/go.uber.org/atomic#readme-usage